src/api/goods.ts

```diff
@@ -3,17 +3,59 @@ import { Good } from '../types/Good';
 // eslint-disable-next-line
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
-export function getAll(): Promise<Good[]> {
+export const getAll = async (): Promise<Good[]> => {
   return fetch(API_URL)
     .then(response => response.json());
-}
+};
+
+export const get5First = async () => {
+  const goods = await getAll();
 
-export const get5First = () => {
-  return getAll()
-    .then(goods => goods); // sort and get the first 5
+  return goods.slice(0, 5);
 };
 
-export const getRedGoods = () => {
-  return getAll()
-    .then(goods => goods); // get only red
+export const getRed = async () => {
+  const goods = await getAll();
+
+  return goods.filter((el: any) => el.color === 'red');
 };
+
+// --- variant 2 ----
+// ------------------------------------------------------------
+// export const getAllTodosApi = async () => {
+//   const response = await fetch(API_URL);
```

Comment:
That's great that you figured out with `async await` and `.then` :+1: 
But leave only one variant, please 🙂

Reply:
<img width="612" alt="Screenshot 2022-11-23 at 22 06 52" src="https://user-images.githubusercontent.com/95155260/203567406-bc42955a-8b08-4ea0-bb38-7f10b7494be9.png">
<img width="662" alt="Screenshot 2022-11-23 at 22 06 49" src="https://user-images.githubusercontent.com/95155260/203567425-682b18d5-a4b1-4414-8352-7156aea21268.png">
<img width="623" alt="Screenshot 2022-11-23 at 22 06 56" src="https://user-images.githubusercontent.com/95155260/203567477-7028eda5-1504-4202-9804-282f857633d8.png">